src/test/ui/break-diverging-value.rs

```diff
@@ -0,0 +1,38 @@
+#![feature(never_type)]
+
+fn loop_break_return() -> i32 {
+    let loop_value = loop { break return 0 }; // ok
+}
+
+fn loop_break_loop() -> i32 {
+    let loop_value = loop { break loop {} }; // ok
+}
+
+fn loop_break_break() -> i32 { //~ ERROR mismatched types
+    let loop_value = loop { break break };
+}
+
+fn loop_break_return_2() -> i32 { //~ ERROR mismatched types
+    let loop_value = loop { break { return; () } };
```

Comment:
Can you change this to `break { return 0; () }`?

I think in this case I expect `// ok`, same as `break return 0`, do you agree?

Reply:
I agree. However, apparently `break { return 0 }` (and your example) behave differently than `break return 0`, so I shall have to investigate.

Reply:
I believe this is because of the point that @eddyb raised earlier.

Reply:
See [this comment](https://github.com/rust-lang/rust/pull/77317/files#r510467463)

Reply:
That did fix it, thanks both. I added the test and it's working as expected now.